packages/create-app/templates/default-app/packages/app/public/oauth2-redirect.html

```diff
@@ -0,0 +1,80 @@
+<!doctype html>
```

Comment:
Not too keen on adding this to the app and template tbh. It's a quite specific thing required by one plugin, so I'm thinking it would be better to package up a bit better with the api-docs plugin. Let's say we bump the swagger version and need to make changes to this file, that becomes a migration rather than a simple version bump since this file is decoupled.

I'm thinking we should explore the options of simply moving this to the main app as a component extension, ending up with this in the app:

```tsx
<Route path="/oauth2-redirect.html" element={<ApiDocsSwaggerRedirectHandler />}/>
```

Another option would be to move it to being served by a backend, similar to how the `auth-backend` works, which is a bit of a hassle though since we would have to add a backend plugin.

Going with a frontend solution we could even make it a more generic page that's able to detect what kind of auth flow it is, avoiding extra page components if we want to support additional providers, basically `ApiDocsAuthHandler` instead of `ApiDocsSwaggerRedirectHandler` or something of that sort.

If the api-docs plugin as a required top-level page we can also piggy back on that, letting the redirect be something like `/api-docs/auth-handler`, with `/auth-handler` being a subroute of `/api-docs` that we configure swagger to point to instead of `/oauth2-redirect.html`

Reply:
Not sure if it's worth to implement code around this. But I guess it would quite easy to copy the code from the file into a component. Might be difficult to keep up to date, but you are right, at least it's possible to migrate it.

On the other hand I'm not sure if there are a lot of users that need this functionality at all. I think it's good to know that it works and would dedicate some lines in the plugin README to it. Maybe we can go with the later for now and keep the issue with the idea for creating such a route?

Reply:
Yep, think that makes sense

Reply:
This is why I was hoping to:
1. Get upstream swagger-api-react to include the missing file. It is present in swagger-api and I believe is mistakenly omitted from the react flavor as it seems to "just work."  It has no UI components; it's a 100% script, so react vs. not doesn't seem to matter.
2. Failing that, bundle it with plugin-api-docs. (Or add a dependency for the plain flavor of swagger-api which does include the file.)
3. Add a postinstall script to plugin-api-docs to install the file in /app/public.

I don't think there's any point in making it generic, because it is not a visible page, and that would really become a maintenance nightmare. It is simply used as a callback catcher to pass the OAuth2 `code` to the swagger-api.

BTW, Authorization Code flow is the most popular (only) OAuth2/OIDC mechanism for end-user login so I expect this code sees a lot of use.

I am not knowledgeable about JS/TS, React, nor the structure of Backstage to know how to actually do any of this but am willing to learn.

Reply:
@n2ygk Thoughts on going with just the readme-based solution for now? Linking out to https://github.com/swagger-api/swagger-ui/blob/master/dist/oauth2-redirect.html and telling people to copy that to `public/oauth2-redirect.html` in the app? I feel it's a bit too niche for it to be added to the default app template, and it also creates some obscure coupling.

On top of that we could create an issue around the discussion here, options for shipping it with the plugin rather than having to have people copy it over.

Reply:
I'll push a commit in a bit.

Reply:
Updated only the README.md. 

@Rugvip I don't believe this requires a changeset as there's no code changes. Let me know if that's not correct.